namespaces/mson-namespace.md

```diff
@@ -0,0 +1,496 @@
+
+[MSON][] is used throughout this document.
+
+## Expanded Element
+
+MSON is built around the idea of defining recursive data structures. To provide abstraction, for convenience reasons and to not repeat ourselves, these structures can be named (using an _identifier_) and reused. In [MSON][], the reusable data structures are called _Named Types_.
+
+Often, before an MSON DOM can be processed, referenced _Named Types_ have to be resolved. Resolving references to _Named Types_ is tedious and error prone. As such an MSON processor can resolve references to produce a complete MSON DOM. That is, a DOM that does not include unresolved references to other data structures. This is referred to as _reference expansion_ or simply _expansion_.
+
+In other words, an expanded element is one that does not contain any _Identifier_ (defined below) referencing any other elements than those defined in MSON namespaces.
+
+The expanded DOM MUST, however, keep the track of what data structure was expanded and what from where.
+
+## Base Element
+
+In MSON, every data structure is a sub-type of another data structure, and, therefore, it is directly or indirectly derived from one of the MSON _Base Types_. This is expressed as an inheritance of elements in MSON DOM. Where the predecessor of an element is referred to as element's _Base Element_.
+
+Note: Not every MSON _Base Type_ is presented in Refract primitive types and vice versa – see the table below.
+
+### Type comparison
+
+| JSON primitive |      Refract     | MSON Base Type | MSON Namespace |
+|:--------------:|:----------------:|:--------------:|:--------------:|
+|     boolean    |  Boolean Element |     boolean    |  Boolean Type  |
+|     string     |  String Element  |     string     |   String Type  |
+|     number     |  Number Element  |     number     |   Number Type  |
+|      array     |   Array Element  |      array     |   Array Type   |
+|        -       |         -        |      enum      |    Enum Type   |
+|     object     |  Object Element  |     object     |   Object Type  |
+|      null      |   Null Element   |        -       |        -       |
+|        -       |  Member Element  |        -       |  Property Type |
+
+# MSON DOM Elements
+
+## MSON Element (Element)
+
+Base element for every MSON element.
+
+The MSON Element adds attributes representing MSON _Type Definition_ and _Type Sections_.
+
+Note: In MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the element.
+
+### Properties
+
+- `attributes`
+    - `typeAttributes` (array) - _Type Definition_ attributes list, see _Type Attribute_  
+        - (enum[string])
+            - required
+            - optional
+            - fixed
+            - sample
+            - default
+    - `variable` (boolean) - Element content is _Variable Value_
+    - `sample` (*T*) - Alternative sample value for _Member Types_
```

Comment:
Also not really sure about `*T*` as the type – the way it is written is definitely not a valid MSON syntax, also the type of this should be always the type in the `content` of this element (= `MSON Element` ?)